.github/workflows/test.yml

```diff
@@ -0,0 +1,248 @@
+name: Code Build and Tests
+
+on:
+  workflow_call:
+
+permissions:
+  contents: read
+  checks: write
+
+jobs:
+  ts-unit-tests:
+    name: TypeScript Unit Tests
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16.14.2
+          cache: yarn
+          cache-dependency-path: ./internal/lookout/ui/yarn.lock
+
+      - name: Run Unit Tests
+        run: |
+          yarn install --frozen-lockfile && yarn run openapi && CI=true yarn run test --reporters=jest-junit
+        working-directory: ./internal/lookout/ui
+
+      - name: Publish JUnit Report
+        uses: mikepenz/action-junit-report@v3
+        if: always()
+        with:
+          report_paths: ./internal/lookout/ui/junit.xml
+          fail_on_failure: true
+          require_tests: true
+          detailed_summary: true
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Upload Test Reports Artifacts
+        if: always()
+        uses: actions/upload-artifact@v3
+        with:
+          name: ts-unit-test-reports
+          path: ./internal/lookout/ui/junit.xml
+          if-no-files-found: error
+
+      - name: Send Coverage Report to Codecov
```

Comment:
Are we able to test that this produces the results we expect?

Reply:
https://app.codecov.io/gh/armadaproject/armada